src/desktop/apps/auction/routes.js

```diff
@@ -84,6 +84,10 @@ export async function index(req, res, next) {
       sort = "-searchable_estimate"
     }
 
+    const modal = req.originalUrl.match("/confirm-registration")
+      ? "ConfirmRegistration"
+      : null
```

Comment:
Initially I used the same [location check that the backbone implementation used when deciding whether to show the modal](https://github.com/artsy/force/pull/4277/files#diff-aa8dd8a4c5ba63e10645d7a9555718d9L86), but this seemed to cause issues when I did it (in `Layout.jsx`), causing the modal to render with white text inside the auction banner. It was fine when i was not testing (because i wanted to see it while i got the UI right), and I figured this was due to a SSR mismatch which led it to mount incorrectly but didn't pursue it too much farther. Instead I moved it into the server-side redux store preparation. My thought is that this logic can be expanded in the future if we want to choose between multiple modals.

That said it feels awkward and I'm open to other solutions.

Reply:
Is it possible to do this client-side? Would that break SSR?

```ts
 ... = window.location.match("/confirm-registration") ? ... : ...
```

Reply:
That was my initial approach. I could give it one more try today but doing this caused it to render incorrectly.

Reply:
If I change the conditional logic here to
```tsx
  const Modal =
    typeof window !== "undefined" &&
    window.location.pathname.match(/\/confirm-registration$/)
      ? ConfirmRegistrationModal
      : null
  // modal === "ConfirmRegistration" ? ConfirmRegistrationModal : null
```
It renders inside the banner and pushes the actual banner content out. I assume the white text that I highlighted is white because it is inheriting from the banner. Also when I click to close the modal, the entire banner is removed 🙃 
![Screen Shot 2019-07-11 at 5 12 56 PM](https://user-images.githubusercontent.com/9088720/61085859-57690880-a3ff-11e9-8e01-a7ff0b4a59a3.png)